nf_deployments/v1alpha1/nf_deployment_types.go

```diff
@@ -34,153 +57,176 @@ type NFDeploymentSpec struct {
 	NetworkInstances []NetworkInstance `json:"networkInstances,omitempty" yaml:"networkInstances,omitempty"`
 	// configRef defines addiitonal configuration references the nf depends upon
 	// +optional
-	ConfigRefs []corev1.ObjectReference `json:"configRefs,omitempty" yaml:"configRefs,omitempty"`
+	ParametersRefs []ObjectReference `json:"parametersRefs,omitempty" yaml:"parametersRefs,omitempty"`
+}
+
+type ObjectReference struct {
+	// APIVersion of the target resources
+	APIVersion string `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`
```

Comment:
apiversion, kind and name should be mandatory in my view

Reply:
I've modified as suggested @henderiw @josephthaliath